source/raylib/raylib_types.d

```diff
@@ -63,6 +63,54 @@ struct Rectangle
     float height;
     alias w = width;
     alias h = height;
+
+    Vector2 origin() { // Rectangle function exclusive to raylib-d
+        return Vector2(x, y);
+    }
+    
+    Vector2 dimensions() {
+        return Vector2(width, height);
+    }
+
+    alias topLeft = origin;
+
+    Vector2 topRight() const {
+        return Vector2(x:(x + width), y:y);
+    }
+
+    Vector2 bottomLeft() const {
+        return Vector2(x:x, y:(y + height));
+    }
+
+    Vector2 bottomRight() const {
+        return Vector2(x:(x + width), y:(y + height));
+    }
+
+    void opOpAssign(string op)(Vector2 offset) {
+        static assert(op=="+" || op=="-");
```

Comment:
```suggestion
    void opOpAssign(string op)(Vector2 offset) if (op == "+" || op == "-") {
```

Operator overloads should define using template constraints what they do or do not support.